### PR TITLE
Use user project's object version for pods project

### PR DESCRIPTION
**Pull request by neonichu**
Inspects the object version of all user projects and uses the lowest one for the pods project. The default version will be used if there is no user project.

See also CocoaPods/Xcodeproj#253

### CODE REVIEW

**Comment by neonichu** [comment]
cc @segiddins @kylef

**Comment by segiddins** [comment]
:ship:

File: lib/cocoapods/installer.rb [hunk 1]
```diff
@@ -444,7 +444,15 @@ def warn_for_deprecations
     #
     def prepare_pods_project
       UI.message '- Creating Pods project' do
-        @pods_project = Pod::Project.new(sandbox.project_path)
+        object_version = aggregate_targets.map(&:user_project_path).compact.map do |path|
+          Xcodeproj::Project.open(path)
```
Comment (segiddins) [inline lib/cocoapods/installer.rb]: `.obejct_version.to_i`

File: spec/unit/installer_spec.rb [hunk 2]
```diff
@@ -585,6 +589,23 @@ def test_extension_target(symbol_type)
           @installer.pods_project.expects(:save)
           @installer.send(:write_pod_project)
         end
+
+        it 'uses the user project\'s object version for the pods project' do
```
Comment (segiddins) [inline spec/unit/installer_spec.rb]: double quotes

**Pull request merged**